orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -311,6 +311,10 @@ def _export_model(self, *inputs, **kwargs):
         #       Model is not re-exported when the model parameters change. This can happen when the model is a stateful model,
         #       or the user explicitly changed model parameters after the onnx export.
 
+        # record random states here and restore later in case any of them gets changed during the export,
+        # e.g., some sympy functions in symbolic_shape_infer will change Python's random state.
+        random_states = _utils.get_random_states()
```

Comment:
@thiagocrepaldi sorry I did not make it clear: the random state change actually occurs during the [symbolic shape inference](https://github.com/microsoft/onnxruntime/blob/9f5db78aac10699d0dbce6a59010fb53502d1152/orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py#L330-L332) for this model's case. More precisely, it happens within the `_infer_Slice`. I added it here just in case some future change would also touch any of the random states.